pkg/network/setup/netpod/netpod.go

```diff
@@ -106,6 +107,13 @@ func NewNetPod(vmiNetworks []v1.Network, vmiIfaces []v1.Interface, vmiUID string
 	for _, opt := range opts {
 		opt(&n)
 	}
+
+	ifaceStatusesByName := vmispec.IndexInterfaceStatusByName(n.vmiIfaceStatuses, func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
+		return ifaceStatus.Name != ""
+	})
+
+	n.podIfaceNamesByNetName = namescheme.CreateFromIfaceStatuses(n.vmiSpecNets, ifaceStatusesByName)
+
```

Comment:
> I think introducing some "withPodIfaceNameScheme" option funtion would fit better.

This is an interesting idea, could you please elaborate what advantages is has over the proposed solution?

> Please note this change seem to make the controller reconciliation depend on a status.

It makes the virt-handler and virt-launcher depend on information generated by the VMI controller and propagated via the VMI.Status, as was proposed in https://github.com/kubevirt/community/pull/324.

Reply:
> This is an interesting idea, could you please elaborate what advantages is has over the proposed solution?

Current solution blots the factory adding complexity to it https://github.com/kubevirt/kubevirt/blob/677282d3df8db97d5d7c9608b82bfa540397cbf6/pkg/network/setup/netpod/netpod.go#L89-L124.
Alternatively, we could filter ifaces/nets who's `podIfaceName` is reported at the caller.
And pass the name-scheme using an option function.

Reply:
> Current solution blots the factory adding complexity to it

> Alternatively, we could filter ifaces/nets who's podIfaceName is reported at the caller. 
And pass the name-scheme using an option function.

Some one, at some level will have to do it - and it will add a bit of filtering logic to that level.
The callers are:
1. `VirtualMachineController.vmUpdateHelperDefault` and `VirtualMachineController.vmUpdateHelperMigrationTarget`.
2. `VirtualMachineController.setupNetwork`.
3. `NetConf.Setup`.

Which level do you suggest should have the filtering logic?

Reply:
I am aware we cannot avoid the additional filtering logic, my suggestion is moving it out side the factory (NewNetPod) to keep it simple as it is now.
The only place make sense to me to do these logic is the caller of the factory, that is NetConf.Setup.

The problem I see here is the additional code assume the WithVMIIfaceStatuses option is given, in practice it does, but when you look at the factory code its not straightforward to realize where n.vmiIfaceStatuses comes from.

Since its about cosmetic and not actual functionality issue, lets wait to see what over reviewers think.

Reply:
> The only place make sense to me to do these logic is the caller of the factory, that is NetConf.Setup.

Could you please elaborate what will be the advantages of doing so?

> The problem I see here is the additional code assume the WithVMIIfaceStatuses option is given, in practice it does, but when you look at the factory code its not straightforward to realize where n.vmiIfaceStatuses comes from.

The alternative that I currently have in mind is to make `NewNetPod` take another mandatory parameter. Because it already takes seven mandatory parameters and one variadic - I find it unreasonable.

Reply:
> > I find it unreasonable
> 
> I find it reasonable

You need to back this opinion with something.

`NetPod` needs a lot of data in order to operate.
Depending on the scenario, the data may exist or not, but it needs to work either way.

If someone has a suggestion on how to make that object better, please send a PR so we can all take advantage of it.

--

Now, I should not comment on this here, but this may assist someone that wants to follow up on this in a separate PR: As a basic programming concept, a function, even if it is just a factory, should have a reasonable amount of variables, which is probably much less than 7. The building pattern solves this issue nicely, but it surely is not the only option. Other options include splitting the object into many, separating functionality and re-arranging the pieces.

Lets continue this on a different PR, not here.

Reply:
Fair enough.